### PR TITLE
Update gtest and replace GTEST_FLAG ahead of its deprecation

### PR DESCRIPTION
**Pull request by yanavlasov**
Risk Level: Low, test only
Testing: All tests
Docs Changes: N/A
Release Notes: N/A
Platform Specific Features: N/A

Signed-off-by: Yan Avlasov <yavlasov@google.com>

### CODE REVIEW

File: bazel/repository_locations.bzl [hunk 1]
```diff
@@ -585,10 +585,8 @@ REPOSITORY_LOCATIONS_SPEC = dict(
         project_name = "Google Test",
         project_desc = "Google's C++ test framework",
         project_url = "https://github.com/google/googletest",
-        # Pick up fix for MOCK_METHOD compilation with clang-cl for Windows (resolved after 1.10.0)
-        # see https://github.com/google/googletest/issues/2490
-        version = "a4ab0abb93620ce26efad9de9296b73b16e88588",
-        sha256 = "7897bfaa5ad39a479177cfb5c3ce010184dbaee22a7c3727b212282871918751",
+        version = "977cffc4423a2d6c0df3fc9a7b5253b8f79c3f18",
```
Comment (asraa) [inline bazel/repository_locations.bzl:588]: is it easy to update to the latest commit if not using the latest version?

File: bazel/repository_locations.bzl [hunk 2]
```diff
@@ -585,10 +585,8 @@ REPOSITORY_LOCATIONS_SPEC = dict(
...
+        sha256 = "d20a329b648be60383629f4bfa464f08ea06a760ae4c9fde09d84847cc517502",
```
Comment (asraa) [inline bazel/repository_locations.bzl:589]: date in line 592 needs to be changed to 2021-07-07

**Review by asraa — CHANGES_REQUESTED** [review]

**Review by asraa — DISMISSED** [review]
LGTM to me, pass to @envoyproxy/dependency-shepherds

**Comment by mattklein123** [comment]
Looks like a legit CI failure.

/wait

**Comment by yanavlasov** [comment]
Yeah, trying to diagnose why gcc and cl are unhappy.

**Comment by yanavlasov** [comment]
Windows build hopelessly broken. Reaching out to gtest devs to see if there is a fix.

**Pull request closed without merging**